src/custom/state/lists/hooks/hooksMod.ts

```diff
@@ -21,140 +12,52 @@ import {
   removeGpUnsupportedToken,
 } from '../actions'
 import { UnsupportedToken } from '@cow/api/gnosisProtocol'
-import { isAddress } from 'utils'
 import { SupportedChainId as ChainId } from 'constants/chains'
 import { supportedChainId } from 'utils/supportedChainId'
-import { TokenAddressMap, combineMaps } from '@src/state/lists/hooks'
 import { shallowEqual } from 'react-redux'
+import { TokenInfo } from '@uniswap/token-lists'
+import { Currency } from '@uniswap/sdk-core'
 
-/* export type TokenAddressMap = ChainTokenMap
-
-type Mutable<T> = {
-  -readonly [P in keyof T]: Mutable<T[P]>
-}
-*/
-
-/**
- * An empty result, useful as a default.
- */
-export const EMPTY_LIST: TokenAddressMap = {
-  [ChainId.MAINNET]: {},
-  [ChainId.GNOSIS_CHAIN]: {},
-  [ChainId.GOERLI]: {},
-}
-
-const listCache: WeakMap<TokenList, TokenAddressMap> | null =
-  typeof WeakMap !== 'undefined' ? new WeakMap<TokenList, TokenAddressMap>() : null
-
-export function listToTokenMap(list: TokenList): TokenAddressMap {
-  const result = listCache?.get(list)
-  if (result) return result
-
-  /*
-    const map = list.tokens.reduce<Mutable<TokenAddressMap>>((tokenMap, tokenInfo) => {
-    const token = new WrappedTokenInfo(tokenInfo, list)
-    if (tokenMap[token.chainId]?.[token.address] !== undefined) {
-      console.error(`Duplicate token! ${token.address}`)
-      return tokenMap
-    }
-    if (!tokenMap[token.chainId]) tokenMap[token.chainId] = {}
-    tokenMap[token.chainId][token.address] = {
-      token,
-      list,
-    }
-    return tokenMap
-  }, {}) as TokenAddressMap
-    listCache?.set(list, map)
-    return map
-  }
-*/
-
-  const map = list.tokens.reduce<TokenAddressMap>((tokenMap, tokenInfo) => {
-    const token = new WrappedTokenInfo(tokenInfo, list)
```

Comment:
`WrappedTokenInfo` is very heavy class, because:
 - it calculates `checksummedAddress` using `keccak256` (see `getAddress`)
 - it contains lots of unused parameters, like a link to the tokens list (`public readonly list?: TokenList`)